File: Python3/Bit Manipulation/338. Counting Bits.py
""" 338. Counting Bits
Difficulty: Medium
Submiission Link: https://leetcode.com/problems/counting-bits/submissions/1037894519/
 """

class Solution:
    def countBits(self, n: int) -> list[int]:
        sum=0
        list1 = []
        for i in range(n+1):
            sum=bin(i).count("1")
            list1.append(sum)
        return list1

File: C/Bit Manipulation/338. Counting Bits.c
/* 338. Counting Bits
Difficulty: Medium
Submiission Link: https://leetcode.com/problems/counting-bits/submissions/1037892747/ */

public class Solution
{
    public int[] CountBits(int n)
    {
        int[] list1 = new int[n + 1];
        for (int i = 0; i <= n; i++)
        {
            int sum = CountOnesInBinary(i);
            list1[i] = sum;
        }
        return list1;
    }

    private int CountOnesInBinary(int num)
    {
        int count = 0;
        while (num > 0)
        {
            count += num & 1;
            num >>= 1;
        }
        return count;
    }
}